### ISSUE TITLE
Example relation widget preview no longer works on first load.

### ISSUE THREAD

**Issue by tech4him1** [netlify/netlify-cms#1541]
**- Do you want to request a *feature* or report a *bug*?**
bug

**- What is the current behavior?**
The kitchen sink custom relation widget preview no longer displays on first load, only after the relation is edited. This may be indicative of a larger problem.

**- If the current behavior is a bug, please provide the steps to reproduce.**

**- What is the expected behavior?**

**- Please mention your versions where applicable.**

Netlify CMS version:
Browser version:

Node.JS version:
Operating System:

**- Please link or paste your `config.yml` below if applicable.**

**Comment by erquhart** [comment]
We can definitely expand the condition for updating there, I just wasn't aware of any when that change was made (like a year ago I think).

**Comment by tech4him1** [comment]
@erquhart If you have a suggestion for the best way to compare here, I'd be happy to implement.

**Comment by erquhart** [comment]
Maybe try checking for changes in the `metadata` prop?

**Comment by tech4him1** [comment]
@erquhart The unique case of the relation widget is that it uses information from other widgets, not just itself, so it would have to update when every other widget updates as well, which could make it more complicated.

**Comment by erquhart** [comment]
Dug into this a bit more, I think the core issue is that `PreviewHOC` isn't as evolved as `EditorControl`/`Widget`. Control components can override the `shouldComponentUpdate` method of the wrapper we provide - preview components cannot.

Before we fix this, though, I've been racking my brain over our internal handling of widgets, and I think I'm going to put together an overhaul that flattens the internal structure. That's going to change updating logic on the control side in a big way, so we should hold off on copying that method over to the preview side.

**Comment by charlie-s** [comment]
I'm not sure if this is the issue that I'm currently experiencing. I added a relation widget, but no option shows up unless I either a) blur the field after typing in 3-4 characters and then focus it again, or b) type in the entire relation name. I'm not sure if the 2nd issue is the intended functionality. I'd be happy to work on this if the 1st issue is indeed related.

**Comment by erquhart** [comment]
@charlie-s not necessarily the same issue, but I have noticed that and you're more than welcome to work on a fix. I'd start with opening a new issue for clarity.

**Comment by barthc** [comment]
@erquhart  I suggest will allow the `PreviewHOC` to update on changes to `fieldsmetadata` , until we refactor the widgets handling.

**Comment by erquhart** [comment]
I thought someone had a PR up that did that, but not seeing it right now. I definitely agree, we'd just need to guard against performance issues during implementation.
